core/domain/stats_services.py

```diff
@@ -1024,3 +1025,198 @@ def _get_calc_output(exploration_id, state_name, calculation_id):
             calculation_output)
     else:
         return None
+
+
+def change_learner_answer_details(
```

Comment:
maybe rename to apply_change_list?

Reply:
As I am not applying list here, as its not the case of exploration where changes will be made first and then applied at the end, in this case, the changes will be automatically applied after performing the action. So apply_change_list might be ambiguous, what do you suggest?

Reply:
Okay I corrected the design now.